portfolio/src/main/webapp/script.js

```diff
@@ -101,4 +101,55 @@ function handleResponse(response) {
 function addName(name) {
     const helloContainer = document.getElementById("hello-container");
     helloContainer.innerText = name;
+<<<<<<< HEAD
+}*/
+
+// To parse JSON into a usable object for javascript
+function getComments() {
+
+    // fetch data from server, parse as json file, then reference coms as an object
+    fetch('/data').then(response => response.json()).then((coms) => {
+        const commentList = document.getElementById("comment-container");
+        commentList.innerText = '';
+        commentList.appendChild(
+            createListElement("Page One: " + coms[0]));
+        commentList.appendChild(
+            createListElement("Page Two: " + coms[1]));
+        commentList.appendChild(
+            createListElement("Page Three: " + coms[2]));
+    });
 }
+
+// For learning promises
+function getCommentsLong() {
+    console.log('fetching JSON');
+
+    const responsePromise = fetch('/data');
+
+    responsePromise.then(handleResponse);
+}
+
+function handleResponse(response) {
+    console.log('convert JSON promise stream to text');
+    const textPromise = response.text();
+    console.log(textPromise);
+    textPromise.then(addListToDom);
+}
+
+function addListToDom(list) {
+    console.log('parsing json text and printing' + list);
+    const commentList = document.getElementById("comment-container");
+    commentList.innerText = '';
+    // list.pageOne does not work as a JSON element should, should this work with array format?
```

Comment:
You have two functions that convert the array to json.

This manually one creates a map using `{}` so you should be able to access it via list.pageOne to get the comments[0].
```
private String convertToJson(ArrayList<String> comments) {
    String json = "{";
    json += "\"pageOne\": ";
    json += "\"" + comments.get(0) + "\"";
    json += ", ";
    json += "\"pageTwo\": ";
    json += "\"" + comments.get(1) + "\"";
    json += ", ";
    json += "\"pageThree\": ";
    json += comments.get(2);
    json += "}";
    return json;
  }

```

For the function `convertJsonUsingGson` that using gson that converts the comments array to a json array so you would need to use comments[0] to access the array.